pkg/agent/run.go

```diff
@@ -224,3 +230,30 @@ func updateAddressLabels(agentConfig *daemonconfig.Agent, nodeLabels map[string]
 	result = labels.Merge(nodeLabels, result)
 	return result, !equality.Semantic.DeepEqual(nodeLabels, result)
 }
+
+func setNodeConfigAnnotations(ctx context.Context, agentConfig *daemonconfig.Agent, nodes v1.NodeInterface) error {
```

Comment:
a couple thoughts on this function:
1. Right before we call this function, we call a syncLabels function which _also_ loops and sleeps forever until the kubelet is ready and also updates the node. So, we're doubling our writes to the node on startup
2. syncLabels at least intelligently decides if an update is needed. This will update the node on every restart. Should we minimize writes by hashing the previously saved annotation values and comparing them against the new  values?

Reply:
1. Yeah I think it makes sense to combine both functions, will fix that
2. Okay I will make sure to compare the hash every time before adding the node config

Reply:
fixed